module/mempool/consensus/exec_fork_suppressor.go

```diff
@@ -132,30 +123,15 @@ func (s *ExecForkSuppressor) Add(newSeal *flow.IncorporatedResultSeal) (bool, er
 		return false, nil
 	}
 
-	// STEP 4: check whether wrapped mempool ejected the newSeal right away;
-	// important to prevent memory leak.
-	// Generally speaking this is a valid situation but our implementation of IncorporatedResultSeals
-	// doesn't allow ejected seals so this case shouldn't happen at all.
-	newSealID := newSeal.ID()
-	if _, exists := s.seals.ByID(newSealID); !exists {
-		return added, nil
-	}
-
-	// STEP 4: add newSeal to secondary index of this wrapper
-	// CAUTION: the following edge case needs to be considered:
-	//  * the mempool only holds a single other seal (denominated as `otherSeal`) for this block
-	//  * upon adding the new seal, the mempool might decide to eject otherSeal
-	//  * during the ejection, we will delete the entire set from the `sealsForBlock`
-	//    because at this time, it only held otherSeal, which was ejected
-	// Therefore, the value for `found` in the line below might
-	// be different than the value in the earlier call above.
+	// STEP 3: add newSeal to secondary index of this wrapper
+	// CAUTION: We expect that underlying mempool NEVER ejects seals because it breaks liveness.
```

Comment:
We should add this requirement to the godoc for `IncorporatedResultSeals`:
https://github.com/onflow/flow-go/blob/f7bdce3a8204e108dc05e28c0b75a3104cf81602/module/mempool/stdmap/incorporated_result_seals.go#L14

Reply:
https://github.com/onflow/flow-go/commit/a9c1bf9c5d5e120bb0a5a3022d3e4a08ee82db7c